app/controllers/scenarios_controller.rb

```diff
@@ -95,6 +95,16 @@ def update
     end
   end
 
+  def enable_or_disable_all_agents
+    @scenario = current_user.scenarios.find(params[:id])
+    @scenario.agents.update_all disabled: params[:scenario][:disabled]
+
+    respond_to do |format|
+      format.html { redirect_to @scenario }
+      flash[:notice] = 'The agents has been successfully updated.'
+    end
+  end
+
```

Comment:
Don't worry about the code complexity offenses of rubocop too much, in my suggestion #1505 I disabled them completely because the default values are really low and almost impossible to achieve in a rails project (at least in my experience) 😄